src/rosdistro/freeze_source.py

```diff
@@ -82,12 +82,32 @@ def freeze_distribution_sources(dist, release_version=False, release_tag=False,
         print("")
 
 
-def _worker(work_queue):
+# Get the repo commit information
+def _get_repo_info(source_repo, retry_count=3, sleep_time=1):
```

Comment:
The current argument `retry_count` is misleading imo. I would expect that after the first attempt it will be *retried* N times but that doesn't seem to be the case. It is only being retried N-1 times for an overall number of N attempts.

Therefore I would suggest to change the logic so that it matches the meaning of "retry". Passing `0` should try once, passing `1` should try up to two times, etc.

Reply:
Good catch. I had changed it so that the error message made more sense vs having it say Attempt 0 of x.  I've increased the retry count to account for the initial attempt.